frontend/pages/queries/details/QueryDetailsPage/QueryDetailsPage.tsx

```diff
@@ -201,11 +201,22 @@ const QueryDetailsPage = ({
   const isApiError = storedQueryError || queryReportError;
   const isClipped =
     (queryReport?.results?.length ?? 0) >= QUERY_REPORT_RESULTS_LIMIT;
-  const disabledLiveQuery = config?.server_settings.live_query_disabled;
+  const isLiveQueryDisabled = config?.server_settings.live_query_disabled;
 
   const renderHeader = () => {
+    // Team admins/maintainers can only edit queries assigned to a team
     const canEditQuery =
-      isGlobalAdmin || isGlobalMaintainer || isTeamMaintainerOrTeamAdmin;
+      isGlobalAdmin ||
+      isGlobalMaintainer ||
+      (isTeamMaintainerOrTeamAdmin && storedQuery?.team_id);
```

Comment:
just to double check, does this guarantee that it's for the same team as the `storedQuery?.team_id`?

Reply:
Let me know if the comment I left in the code is not clear enough:
>     // Team admins/maintainers can only edit queries assigned to a team